public/src/components/Home/ArchivedSubredditsComponent.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import ArchivedSubredditItem from './ArchivedSubredditItem';
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class ArchivedSubredditsComponent extends React.Component {
+    getErrorMessages() {
+        const { errorMessages } = this.props;
+        return errorMessages.map((message, index) => (
+            <div className="errorMessage" key={index}>{message}</div>
+        ));
+    }
+
+    getArchivedSubredditsList() {
+        const {
+            subreddits,
+            archiveFormValue,
+            storeSubredditToArchive,
+            removePosts,
+        } = this.props;
+
+        const subredditList = [...subreddits].map((subreddit, key) => {
+            const formValue = archiveFormValue.toLowerCase().trim();
+            const subredditTitle = subreddit.display_name.toLowerCase().trim();
+
+            if (subredditTitle.search(formValue) > -1 && subreddit.isArchived) {
+                return (
+                    <ArchivedSubredditItem key={key}
+                                           storeSubredditToArchive={storeSubredditToArchive}
+                                           removePosts={removePosts}
+                                           subreddit={subreddit}
+                    />
+                );
+            }
+            return null;
+        });
+
+        if (subredditList.some(subreddit => subreddit)) {
```

Comment:
I think you can do `.filter(Boolean)` after map or use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap and then do `if (subredditList.length)`